src/python/CMSMonitoring/StompAMQ.py

```diff
@@ -41,20 +42,18 @@ def validate(doc, schema='auto', verbose=False):
                 _local_schemas[schema] = json.load(open(schema))
                 logging.warning('Successfully loaded local schema {} for validation'.format(schema))
             except ValueError:
-                logging.warning('Local schema {} is not json compliant'.format(schema))
+                logging.error('Local schema {} is not json compliant'.format(schema))
 
     if schema in _local_schemas:
-        return validate_schema(_local_schemas[schema], doc, verbose)
-
-    elif schema in _schemas.schemas():
-        return validate_schema(_schemas.schemas()[schema], doc, verbose)
+        return validate_schema(_local_schemas[schema], doc, loglevel)
 
     else:
-        for sname, sch in _schemas.schemas().items():
-            if validate_schema(sch, doc, verbose):
-                return True
+        for sch in _schemas.schemas():
+            if schema in [sch, os.path.basename(sch).rsplit('.')[0]]:
```

Comment:
I would replace it with
```
if sch.find(schema) != -1
```
which basically string operation on sch name to find a sub-string schema in it.

Reply:
That would be equivalent to `if schema in sch`, no? I went with the explicit approach because otherwise you get tricky bugs when people name their schemas "JobMonitoring.json" and "JobMonitoringExtra.json" and the "JobMonitoring" string will hit both.